### PR TITLE
fix: add missing `Vulnerability.properties` types in schema 1.4

### PR DESCRIPTION
**Pull request by desenna**
Closes https://github.com/CycloneDX/specification/issues/147

### CODE REVIEW

File: tools/src/test/resources/1.4/valid-vulnerability-1.4.xml [hunk 1]
```diff
@@ -116,6 +116,12 @@
                     </versions>
                 </target>
             </affects>
+            <properties>
+                <property name="Foo">Bar</property>
```
Comment (jkowalleck) [inline tools/src/test/resources/1.4/valid-vulnerability-1.4.xml:120]: your currently suggested XSD does not specify any children inside the  `properties` element

**Review by jkowalleck — CHANGES_REQUESTED** [review]
see my annotations.
please have them sorted out

File: schema/bom-1.4.xsd [hunk 2]
```diff
@@ -2014,6 +2014,16 @@ limitations under the License.
                     </xs:sequence>
                 </xs:complexType>
             </xs:element>
+            <xs:element name="properties" type="bom:propertiesType" minOccurs="0" maxOccurs="1">
```
[the remark this answers is not in the document]
Comment (desenna) [reply schema/bom-1.4.xsd:2017]: Empty element here just means that `properties` here is optional, which aligns with the spec IIUC and is the same (as in copy-paste) as the other usages of `properties` in this file e.g. https://github.com/CycloneDX/specification/blob/c3da323e637e4fbe770126ce3359faa9770f8743/schema/bom-1.4.xsd#L90-L99

Here is the published spec for JSON 1.4 which says "Formal registration is OPTIONAL.": https://cyclonedx.org/docs/1.4/json/#vulnerabilities_items_properties

File: tools/src/test/resources/1.4/valid-vulnerability-1.4.xml (hunk 1 above)
Comment (desenna) [reply tools/src/test/resources/1.4/valid-vulnerability-1.4.xml:120]: IIUC the XSD is saying that adding properties of type "bom:propertiesType" is optional.

Maybe I'm misunderstanding, but in the same way that [this `valid-properties-1.4.xml` mirrors that of its JSON counterpart](https://github.com/CycloneDX/specification/blob/c3da323e637e4fbe770126ce3359faa9770f8743/tools/src/test/resources/1.4/valid-properties-1.4.xml#L4-L9), I thought it was important to bring this to `properties` snippet to this vulnerability _test_ file in order to verify that this XML complies with the updated schema. Without the corresponding XML schema change, [we know from this PR that an error that the XML will not comply and thus error](https://github.com/CycloneDX/cyclonedx-go/pull/39), which gave rise to the issue that this PR is designed to close.

Does this make sense..? I could very well be wrong as I don't know how to run tests / develop locally in this repo (didn't see it in the README), but if you wouldn't mind sharing with me how to do that, I'd be happy to do that and post test results. (I think it'd be a good addition to the README anyhow which I don't mind doing.)

**Comment by desenna** [comment]
> see my annotations. please have them sorted out

@jkowalleck Thanks for the review. I responded to the comments with explanations, but don't understand what's wrong with them. Please feel free to elaborate on what specifically I'm getting wrong and I'll be happy to fix them. Also, if you know how to develop / test locally, can you please share that info with me so I can do? It's possible / likely that I made a mistake here somewhere, but I didn't see how to run tests in the instructions and afaik CI hasn't kicked off fully here so I can't see what steps are taken to setup and run tests. Thanks again!

File: schema/bom-1.4.xsd (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkowalleck) [reply schema/bom-1.4.xsd:2017]: got it. 
/closed

File: tools/src/test/resources/1.4/valid-vulnerability-1.4.xml (hunk 1 above)
Comment (jkowalleck) [reply tools/src/test/resources/1.4/valid-vulnerability-1.4.xml:120]: got it. 
/closed

**Comment by jkowalleck** [comment]
@stevespringett  can kick off the CI.

**Review by jkowalleck — APPROVED** [review]

**Comment by desenna** [comment]
@jkowalleck @stevespringett I'm not able to understand what is causing the test to fail. I can run it locally but I don't see anything except that false is not true. Do you have any advice on how to proceed with this?

**Comment by stevespringett** [comment]
@desenna don't worry about the failure. The CycloneDX Core Java library is the thing that tests it, and it uses local schemas, not the updates you've submitted.

**Comment by desenna** [comment]
> @desenna don't worry about the failure. The CycloneDX Core Java library is the thing that tests it, and it uses local schemas, not the updates you've submitted.

Thanks @stevespringett makes sense.

So are there any other steps I should take here? Or is it ready to merge and deploy you think? Asking only because I can then get to work on resolving https://github.com/CycloneDX/cyclonedx-go/issues/40

Thanks and have a great weekend!

**Comment by desenna** [comment]
> @desenna Although this is a defect, it will have a negative impact on everyone using v1.4 at the moment. In order for this to move forward, the core working group would need to vote on incorporating this into v1.4.2 of the XML schema, or wait until v1.5 next year. The next Core Working Group meeting is on Monday June 6. I'll bring this topic up then.

SGTM! I agree this is a good idea to bring up during Core Working Group meeting and that a v1.4.2 patch version makes sense here. 

Another option might even be to leave it as v1.4.1 (if that's where it's currently at) since this doesn't represent a breaking or backwards-incompatible change but instead _completes_ the specification for XML and protobuf.

This carries the side benefit of also giving other library maintainers - like cyclonedx-go and others - the room/time to make changes in those libraries as well, which their consumers would be able to leverage or even make the open source contributions themselves!

**Comment by stevespringett** [comment]
Ok this will be merged in later this week, but not released yet. We will be updating all official libraries with the updated schema and releasing all updates around the same time.

**Comment by desenna** [comment]
> Ok this will be merged in later this week, but not released yet. We will be updating all official libraries with the updated schema and releasing all updates around the same time.

HI @stevespringett just to confirm is this the fix for java? https://github.com/CycloneDX/cyclonedx-core-java/pull/208

**Comment by stevespringett** [comment]
Confirmed. Had vacation, then COVID, but back at it now. Will merge in the next few days.

**Pull request merged**